### ISSUE TITLE
[Miniflare 3] Enforce subrequest limit

### ISSUE THREAD

**Issue by frandiox** [cloudflare/workers-sdk#4359]
Hi 👋 
This is likely a feature request for workerd but I'm opening the issue here because we are only using Miniflare directly. Let me know if I should transfer it to the workerd repo.

In Miniflare 2 we had [`process.env.MINIFLARE_SUBREQUEST_LIMIT`](https://github.com/cloudflare/miniflare/blob/f919a2eaccf30d63f435154969e4233aa3b9531c/packages/shared/src/context.ts#L18) to set a subrequest limit but afaik there's nothing like that in Miniflare 3.

From what I was able to find, workerd could support limit enforcements but it sets them to "null" on purpose ([source](https://github.com/cloudflare/workerd/blob/5525be752933bcd812efdb6d4dbc1b578cf6dfca/src/workerd/server/server.c%2B%2B#L1752-L1768)). There seem to be [concern around limits](https://github.com/cloudflare/workerd/issues/49) due to how they will behave in different platforms. However, I don't think that concern applies to subrequest limit so it would be great if we can enforce it.

Thanks!

**Comment by rschristian** [comment]
Just ran into this as a bit of a surprise myself -- local dev was just fine, but after deploying (on the free tier), I'm hitting the subrequest limit. Would've been nice to know about this ahead of time (though it's my fault for not skimming the usage limits).

**Comment by tobad357** [comment]
It is also an issue doing D1 dev as queries count towards the limits. Would be great if local env enforced this

**Comment by johtso** [comment]
Right, definitely seems like request limits should be enforced locally.. otherwise that's a pretty major difference in behaviour when running in production.

**Comment by maxwellpeterson** [comment]
In addition to subrequest limits, it would be nice to support CPU and memory limits too, even if they're approximate.

**Comment by penalosa** [comment]
I'm closing this as a duplicate of https://github.com/cloudflare/workerd/issues/1291 (which is where this would be implemented), and I've forward that issue to the relevant internal team.
